content/using/running/minio.md

```diff
@@ -0,0 +1,143 @@
+
+Choose 'buckets' from the left-hand menu, and then 'create bucket' at the top of the page. Enter your bucket name (this MUST match the name in your DNS record, e.g. 'media').
+
+### Configure your ship
+
+Head over to Landscape and navigate to the S3 storage setup page at System preferences > Remote Storage, and enter your domain (with protocol) under endpoint, e.g. `https://example.com`. Enter your username and password from step 1 under access key and secret, and then enter the name of the bucket. When the bucket name is combined with the endpoint, you get your bucket URL e.g. `https://media.example.com`.
+
+You can also configure these settings through dojo as shown [here](/using/os/s3).
+
+### That's it!
+
+You should now be able to upload content using your self-hosted MinIO installation.
+
+## Troubleshooting
+
+Landscape chat will fail silently if it cannot connect to your S3 endpoint to upload media. To get an idea of what's going wrong, open the network tab of your browser dev tools, and observe the request when you try and upload media. You should see a failed request, hopefully with an error code or reason for failure.
+
+- If you see a `mixed-content` error, this means that not every part of the set up is using TLS. Most browsers will refuse to load non-HTTPS content from a secure page.
+- If you see a `502 Bad Gateway` error, caddy is unable to reach your MinIO installation. Check MinIO is running and your `reverse_proxy` URLs are correct.
+- If you get a `Permission denied` error, it's likely that your bucket endpoint is incorrect. Ensure your Caddyfile includes the `header_up` options, and that you passed the `MINIO_DOMAIN` variable when running MinIO - otherwise it will default to using the path URL format, which Urbit does not support.
+
+A good way to test your setup is to `curl` your S3 bucket endpoint (not your root S3 endpoint) and see what response you get. For example, if we have a bucket named 'media':
+
+```
+curl https://media.example.com
+```
+
+You should get an XML response listing the contents of your bucket.
+
+## Running MinIO and Urbit on the same machine
+
+You may wonder how it's possible to run Urbit alongside our MinIO set up if they both need ports 80/443. The answer is to proxy Urbit through caddy, exactly the same way as MinIO. You can create as many directives in your Caddyfile as needed, each pointing to a different port.
+
+For example, you could have 3 domains:
+
+- `ship-name.example.com` - proxy to port 8080 where Urbit is running,
+- `s3.example.com` - proxy to port 9001 where your MinIO admin is running, and
+- `media.example.com` - proxy to port 9000 where your MinIO bucket is accessible
+
+Currently, there is no way to specify the HTTP port Landscape runs on, but if 80 is not available at start-up it will try 8080 next. So start caddy first, and when you boot your ship it should detect that port 80 is in use and use 8080 instead.
```

Comment:
the alternative haskell-based urbit binary `urbit-king` included with the ordinary one can specify the port like `urbit-king run -p <ames-port> --http-port <http-port> /path/to/pier`, but otherwise yeah just reverse proxy to 8080